packages/backend/src/database/entities/jobs.ts

```diff
@@ -0,0 +1,43 @@
+import { JobStatusType, JobStepStatusType, JobStepType } from 'common';
+import { Knex } from 'knex';
+
+export const JobsTableName = 'jobs';
+export const JobStepsTableName = 'job_steps';
+
+export type DbJobs = {
+    job_uuid: string;
+    project_uuid: string | undefined;
+    created_at: Date;
+    updated_at: Date;
+    job_status: JobStatusType;
+};
+
+type UpdateJob = Pick<
+    DbJobs,
+    'project_uuid' | 'job_uuid' | 'job_status' | 'updated_at'
+>;
+
+export type JobsTable = Knex.CompositeTableType<DbJobs, UpdateJob>;
+
+export type DbJobSteps = {
```

Comment:
We need a column to store the error if the step fails, or is that something that would live under the step label?

Reply:
I can rename step_label to `step_error : string | undefined` if you want to store errors but don't want to store labels and prefer to generate on the fly